### PR TITLE
UCT/API: enhance doc for the uct_listener functions.

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

**Comment by alinask** [comment]
Adresses https://github.com/openucx/ucx/issues/4040

**Comment by alinask** [comment]
@yosefe @shamisp @tonycurtis Can you please review?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11832/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 18-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/06676a54a42893a963b46a598c3aeeb9680e9684/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/320aa01833abf2d10c6a5f6105577cb97324958a/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/e1a9fd21d7949d502d05af18612e7bfde33e80bd/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/0c6c91f46377cd35510eb926291177b484a24eed/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/d0f528161caeefcfad44079f9b34b60f75f030e3/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/483c5d2892ff848d2cd43395ee9faec69251a29b/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/a0e59cb9c7a3844697433588d6e474bffee25d4d/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/0d6023352f90490ebc566534f22e0dbb94cdae67/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/8403a2cfe895568bb21d44ee7d0306637f1aa3f5/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/706954985931282230f7782541601c1a5bac93f7/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/321aee671f27d23ad69cea26e33d9b5d37779d67/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/c00b17490ff1cbf444ffb68d0146bb5243fdf83c/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/167cdfea11171f3640fc57356342cccea28f6700/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/9632ffe7b6b37a871a0d27560b7a0386a97d0a30/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/ebe7d25fbfb84728116b9b4d5da2103318705a07/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/fb55b3d70b796fe5d5424a89a5d04b0605247f04/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/71acce971201c3a33d289e7c011adc943c4d771c/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/d0b490c4d51a07e75add01a34485dcf0da88f30f/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/05f2c14b787277e022a64dcfe9586fc5fcb7ee10/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/de612dcfd6f024f0f553d085668e9d9ccb7338fd/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/85caeb99ea7fd93efcc2826d2105b486b38e5357/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/f97f85716eeaf12db21407e37b753f0bc067dff2/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/10f83ae2c2111a8567fdb17e1e9a10d394cb49b5/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/0d71c922c00d07e93fbedaa5670660f49c590c07/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/93a0cb8d450fd491c136d0029550e78dea40d453/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/bb3795b10c5b65e1f0493ad2d484611409516428/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/f7987f41fd4dfb71c797d8480d5d9298be551da8/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/9e0e3f73fb80327e81d39f21b9f8531e28fdd153/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/0d5185dc79395c881c1f6d71eeaf8fd7/raw/c7100095dc2e487ac8f04c0165da6a6b52cc065a/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by alinask** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
bot:pipe:retest

File: src/uct/api/uct.h [hunk 1]
```diff
@@ -3044,6 +3032,9 @@ ucs_status_t uct_cm_query(uct_cm_h cm, uct_cm_attr_t *cm_attr);
  * @ingroup UCT_CLIENT_SERVER
  * @brief Create a new transport listener object.
  *
+ * This routine creates a new transport listener on the given CM which
+ * willl start listening on a given sockaddr.
```
Comment (brminich) [inline src/uct/api/uct.h]: `will`

Comment (shamisp) [reply src/uct/api/uct.h]: `This routine creates a new transport listener` ->`This routine creates a new  listener`

File: src/uct/api/uct.h [hunk 2]
```diff
@@ -3070,11 +3061,32 @@ ucs_status_t uct_listener_create(uct_cm_h cm, const struct sockaddr *saddr,
 void uct_listener_destroy(uct_listener_h listener);
 
 
+/**
+ * @ingroup UCT_CLIENT_SERVER
+ * @brief Reject a connection request.
+ *
+ * This routine is performed by the server side. It rejects a connection request
```
Comment (shamisp) [inline src/uct/api/uct.h]: ` is performed ` -> `can be invoked on the server side`

**Comment by alinask** [comment]
@tonycurtis Can you please take a look as well?

**Comment by tonycurtis** [comment]
Looks ok now

**Comment by alinask** [comment]
@yosefe Can you please review?

**Comment by alinask** [comment]
rnat
bot:bgate:bot

**Comment by alinask** [comment]
bot:bgate:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11870/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 20-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/c5aafba2ead51ecc170dd2725bf216fabab97b03/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/c7c31790a85ea32bc5eedbc9eb612869de77c4ae/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/a1870f661259a3c3853577779f9042188596d535/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/fc2caa8048cc2222dc0f9b0ae8d451b69949e2de/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/7ca9a8334ce1fb6fe6d8e597f92e371705348a0f/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/591217b4cdf3c1310ffe0ed930d27613415323c1/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/5dae68779d16e6e905e2fc32922239d2ca780e85/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/f5ae24c1164caae12bbb94eeb9717eddebeaefa8/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/0a91cf17b357fe363f8aa3b213a44101ffb40339/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/c886c067af5fc1ab2464f4cfe18572bcb0bf5710/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/2ae9f5c7e02a56eff78fe7b0f31607849037b107/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/17fefc14df0296635afc3fc41001c1d2d23655ba/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/023712db704809a6c114c561e7bafbe0627e1bdd/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/4aad9fd8f6e81d9b4a32081bf7484b4d62b6ee30/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/ecf6970ea17c20736dd6a696b03de3f375f3083e/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/10d4a9bb26a7aa286d8c367905acf81ccb35689e/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/bef004ad7ff4147d5875e8431fb24614976ca152/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/b103ec13ef4e1ed70ab6f706a9f5e67a069b7636/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/b8bc68e98d20c0103f3e86cbf5d3944bef09feb5/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/2c1319c89489e344ad72e4966972ec61120f230f/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/6dd9b92e380e18d02fe712fb79e05e6d7d7fee94/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/a0591f58e24663ea72a1aee5e6adeed1ed2aa2c6/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/a38dabda999d3cc42718afa03ee2fa76fd382818/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/dd8d5a42e4adabde22096fefa64d89d6b7405cc6/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/105ac31c0810286b59c12f66878a54d0f3ae66cd/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/cbfdb0f49ea6e5519468982d6b284e3b7473a6b1/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/9699246fc529bb3b0f3f241bd6132bda4ea0e32f/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/d8d73162689a1a90e08d30963f67cb21586295c9/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/d182dec73e30e13b09b64770afeb7225/raw/cde7f921940b7dcc40d183528aa939119a4abecc/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Pull request merged**